packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -740,27 +777,44 @@ export class IntervalCollection<TInterval extends ISerializableInterval> extends
         return this.addInternal(serializedInterval, true, undefined);
     }
 
+    /**
+     * @deprecated - delete by start/end position is deprecated. Please use removeIntervalById.
+     */
     public delete(
         start: number,
         end: number,
     ) {
-        let sequenceNumber = 0;
         if (!this.attached) {
             throw new Error("attach must be called prior to deleting intervals");
         }
 
-        if (this.client) {
-            sequenceNumber = this.client.getCurrentSeq();
+        const id = this.localCollection.createLegacyId(start, end);
+        this.removeIntervalById(id);
```

Comment:
this won't work. you need to look up the interval, and find the id, as the start end at load may not be the start end at delete time as intervals can shift when bound to shared string positions

Reply:
True. That's gross. Note that there could be any number of (start, end) intervals in the tree. Our idea was to let the deprecated delete(start, end) remove just the legacy interval to which we gave a dummy ID because it came over the wire without one. But there could actually be any number of those, if the ID doesn't match the current (start, end). This part might actually be simpler if we didn't mock up an ID in the legacy case and just left it blank.

Reply:
you may need to set another property on legacy intevals, so you can easily distinguish them from intervals with ids, and only allow removing legacy intervals wil start/end

Reply:
replied at the same time. i wouldn't optimize for this method, as we should just remove it in the next release

Reply:
or just add a prefix to the legacy id, so we can identify it easily

Reply:
Assuming we can distinguish legacy intervals from new ones, it seems like we still have the potential for multiple legacy intervals with a given (start, end) if they started out with different mocked-up ID's. We don't have that problem if we just don't give them ID's, since they'll always collapse to one. I don't remember the problem we solved by mocking up ID's.

Reply:
it just made some things easier, as you could always assume there was an id

Reply:
My inclination right now is to let legacy intervals merge to one for a given start, end regardless of Id. (Meaning, have them compare equal to one another.) Then this delete can remove the (at most one) legacy interval.

Reply:
Do we gain anything by having a mocked-up ID that includes the start and end, if it's possible to have multiple intervals in the tree with the same legacy ID but not the same (start, end)? It seems like we want to merge all legacy intervals that have a given (start, end) (since that's the legacy behavior anyway), and we want to have a method that can find the unique legacy interval with (start, end), assuming it exists. So are we maybe better off using the same ID for all legacy intervals, regardless of (start, end), so that merging them as they move around just falls out?

Reply:
ahhh. ok this all makes sense now

Reply:
the problem is it doesn't get us out of supporting intervals without ids. in general it doesn't really matter, and no prod scenarios use this, so it should be fine to break back compat. have we released the code just adding ids? ideally that code gets released and consumed, there is still only a single interval per range, they all get some unique id that's not really used, but this get's persisted is the onlyt thing that matter, so when we roll out multiple intervals per range, and id's per operation all intervals have some id, regardless of what it is